tools: Use npm ci in setup-dev.sh

Devp00l opened the pull request:
### tools: Use npm ci in setup-dev.sh

Currently npm install is used which isn't correct as it doesn't
consider the versions mentioned in the `package-lock.json`,
`npm ci` however does.

This also updates the git workflow in order to test the right command.

Signed-off-by: Stephan Müller <smueller@suse.com>

---

### glass: Lock file version change
    
Fundamental changes in the lock file in order to use node 14 and older and not only 16.
    
This change also updates the packages.
    
Signed-off-by: Stephan Müller <smueller@suse.com>


---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test tumbleweed`
- `jenkins run tumbleweed`
- `jenkins test leap`
- `jenkins run leap`

</details>

s0nea commented:
jenkins test tumbleweed

Devp00l commented:
Hm.. seems like the npm version for leap and tumbleweed is to low. `npm ci` currently only works with node 16.x ... Not sure if this is an problem with our package lock file because it was updated at some point with node 16.x

kshtsk commented:
I guess we need to use 14.x version since not all tools support 16.x yet.

votdev commented on tools/setup-dev.sh:
```diff
@@ -363,6 +363,6 @@ deactivate
 
 pushd src/glass &>/dev/null
 rm -rf node_modules
```
You can remove this line because `npm ci` deletes this directory also.

Devp00l replied:
Thanks for the hint :)

Devp00l commented:
Updated the PR description as I changed some things in order to make ``package-lock.json`` usable by older node versions, which is also tested now by the git hub actions.

votdev reviewed (approved):
LGTM

kshtsk commented:
jenkins test tumbleweed

Devp00l commented:
@kshtsk which version is used in the jekins job as two nodejs versions are installed I assume the default node path will use node in version 5 (not sure but it comes with the packet ``nodejs-common-5.0-2.1.x86_64``) instead of 16 which is also installed. I tested it with node 14 up to 16 which worked fine.

kshtsk commented:
> @kshtsk which version is used in the jekins job as two nodejs versions are installed I assume the default node path will use node in version 5 (not sure but it comes with the packet `nodejs-common-5.0-2.1.x86_64`) instead of 16 which is also installed. I tested it with node 14 up to 16 which worked fine.

The version of nodejs can be found in the logs of the tests, also the way it is tested is well defined in .wasser/config.yaml what exactly installed on your system and how do you test it is unclear to me. So I don't know how did you test it with 16 if cypress does not support the 16?

kshtsk commented:
@Devp00l if you take a look at leap run logs, you can find that there is only nodejs installed which is 14.

kshtsk commented:
jenkins run tumbleweed

kshtsk commented:
The problem in regard which nodejs is taken on tumbleweed, because today it can be one version as default, another day it can be changed to another.

Devp00l commented:
Seems to have fixed itself ;)
